### ISSUE TITLE
Add support for ArcGIS Online Search

### ISSUE THREAD

**Issue by ajturner** [koopjs/koop-opendata#9]
This would be extremely powerful if it (or better yet another plugin) called to ArcGIS online. 

`/arcgis-online/search/FeatureServer/0/query?text=crime&geometry=..."

that would be a huge boon to people that want a map search to Online.

**Comment by dmfenton** [comment]
Yeah, I definitely see this as it's own provider. The structure of the code will be the same, but we'll have to change a lot of the method implementations. Should be pretty Straightforward :tm:. I'll take a look at it in the next few weeks.
